### ISSUE TITLE
WebUI login fails when trying to use keystone

### ISSUE THREAD

**Issue by leifmadsen** [cloudwan/gohan#49]
I'm trying to setup authentication with the WebUI via keystone, but it just fails with a `Error:Data Not Found`.

When starting the server, a connection to Keystone is successful (if I use the wrong password for example, I get a CRITICAL and gohan fails to start; when everything is correct, gohan starts as it should).

Here is the console output when trying to login to gohan with Keystone enabled:

```
14:57:44.666 gohan.server.middleware INFO  Started POST /v2.0/tokens for client 127.0.0.1:40542 data: {"auth":{"passwordCredentials":{"username":"demo","password":"welcome"},"tenantName":"demo"}}
14:57:44.666 gohan.server.middleware DEBUG  Request headers: map[Content-Length:[93] Accept:[application/json, text/javascript, */*; q=0.01] Origin:[https://localhost:9443] User-Agent:[Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/46.0.2490.86 Safari/537.36] Content-Type:[application/json] Dnt:[1] Cookie:[org.cups.sid=c524dea0ac5489f42457c22ff312ab19; _ga=GA1.1.279487755.1432391029] Connection:[keep-alive] Referer:[https://localhost:9443/webui/] Accept-Encoding:[gzip, deflate] Accept-Language:[en-GB,en-US;q=0.8,en;q=0.6] X-Requested-With:[XMLHttpRequest]]
14:57:44.666 gohan.server.middleware DEBUG  Request body: {"auth":{"passwordCredentials":{"username":"demo","password":"welcome"},"tenantName":"demo"}}
14:57:44.666 gohan.server.middleware DEBUG  Response headers: map[Access-Control-Allow-Origin:[*] Access-Control-Allow-Headers:[X-Auth-Token, Content-Type] Access-Control-Allow-Methods:[GET,PUT,POST,DELETE] Content-Type:[text/plain; charset=utf-8] X-Content-Type-Options:[nosniff]]
14:57:44.666 gohan.server.middleware DEBUG  Response body: 404 page not found
14:57:44.666 gohan.server.middleware INFO  Completed 404 Not Found in 364.995µs

```

And here is the relevant connection information:

```
# keystone configuraion
keystone:
    use_keystone: true
    fake: false
    auth_url: "http://localhost:35357/v2.0"
    user_name: "admin"
    tenant_name: "admin"
    password: "welcome"
    version: "v2.0"
```

On start up, if I use `tshark` to look at the HTTP traffic, I see a successful authentication. However, when logging into gohan, I don't see any attempt to connect Keystone at all.

**Comment by nati** [comment]
Sorry.  I found doc is missing for this.

You need to set correct AUTH_URL in config.json.

This is sample.
https://github.com/cloudwan/chazuke/blob/master/config.json.sample

**Comment by leifmadsen** [comment]
@nati hrmmmm thanks, I did see that earlier, and now I've updated it, but it still doesn't attempt to send an auth request to keystone.

**Comment by leifmadsen** [comment]
OK I think I figured it out. I enabled the javascript debugger in the browser (your comments gave me the hint). At first I was mixing HTTP (keystone) and HTTPS (browser) and it didn't like that. Once I set everything to the same in my test environment I got a different error.

It looks like the GUI is expecting CORS stuff, and I didn't set that up on the Keystone side (yet) so I'm pretty sure that's what I'm running into now.

There is likely some good docs here that could be enhanced to show how to setup Keystone with the webUI :)

**Comment by nati** [comment]
Thanks!  I will add some doc for this

**Comment by nati** [comment]
added a sample https://github.com/cloudwan/gohan_apps/tree/master/keystone_sample

**Comment by leifmadsen** [comment]
Awesome! I actually figured it out myself too. There is another way to do it if you're just doing Keystone (I'm not using the full OpenStack stack).

With Keystone, I followed this blog post:  https://ianunruh.com/2014/11/openstack-cors.html

Once I did that, I was able to authenticate after running some commands against Keystone per: http://docs.openstack.org/juno/install-guide/install/yum/content/keystone-users.html
